gen3-cli/tests/integration/backend/test/test_generate.py

```diff
@@ -39,9 +42,30 @@ def test():
         assert "test_invalidSyntax():" in file_text
 
     # testing dir scan generation
-    with tempfile.TemporaryDirectory() as dir:
+    with tempfile.TemporaryDirectory() as temp_dir:
         copied_casefiles = []
         for filename in casefiles:
-            filename = shutil.copy2(filename, os.path.join(dir, os.path.basename(filename)))
+            filename = shutil.copy2(filename, os.path.join(temp_dir, os.path.basename(filename)))
             copied_casefiles.append(filename)
-        assert generate_test_backend(copied_casefiles, None) == generate_test_backend(directory=dir, output=None)
+        assert generate_test_backend(copied_casefiles, None) == generate_test_backend(directory=temp_dir, output=None)
+
+    # testing empty no testcase files error
+    with tempfile.TemporaryDirectory() as temp_dir:
+        with pytest.raises(UserFriendlyBackendException) as einfo:
+            generate_test_backend(directory=temp_dir, output=None)
+        assert str(einfo.value) == "No testcase files found!"
+        empty_testcase_file_path = os.path.join(temp_dir, 'empty-testcase.json')
+        with open(empty_testcase_file_path, 'wt+') as f:
+            json.dump({}, f)
+
+        # testing empty testcase files error
+        with pytest.raises(UserFriendlyBackendException) as einfo:
+            generate_test_backend(directory=temp_dir, output=None)
+        assert str(einfo.value) == "No testcases found!"
```

Comment:
unittest for thing generating unittests from unittests templates
![deeper](https://user-images.githubusercontent.com/131850/72001275-63feda00-3245-11ea-922b-e10003ab6c43.jpg)